test/helpers.lua

```diff
@@ -185,7 +188,11 @@ local function check_cores(app, force)
   local gdb_db_cmd = 'gdb -n -batch -ex "thread apply all bt full" "$_NVIM_TEST_APP" -c "$_NVIM_TEST_CORE"'
   local lldb_db_cmd = 'lldb -Q -o "bt all" -f "$_NVIM_TEST_APP" -c "$_NVIM_TEST_CORE"'
   local random_skip = false
-  local local_tmpdir = tmpdir_is_local(tmpdir_get()) and tmpdir_get() or nil
+  -- Workspace-local $TMPDIR, scrubbed and pattern-escaped.
+  -- "./Xtest-tmpdir/" => "Xtest%-tmpdir"
+  local local_tmpdir = (tmpdir_is_local(tmpdir_get())
+    and tmpdir_get():gsub('^[ ./]+',''):gsub('%/+$',''):gsub('([^%w])', '%%%1')
```

Comment:
Yes, I made that tradeoff because of the app-specific nature of this logic: workspace-local temp dir is not going to be an absolute path, and even if it is, removing the preceding slash isn't going to cause a false positive.

Doing it the robust way requires more gsub() calls. Would be nice to have some `path_scrub()` function but again this is application-specific it didn't make sense as a function, unless one writes a function with various parameters to control the behavior (would be welcome, just out of scope here).